torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -74,6 +47,11 @@ def _parse_fill(
 
         return fill
 
+    def _get_params(self, sample: Any) -> Dict[str, Any]:
+        num_channels, height, width = query_chw(sample)
+        fill = self._parse_fill(image, num_channels)
```

Comment:
ARGH, we need the actual image here due to 

https://github.com/pytorch/vision/blob/857c03031c4bedeea34e489fd9b1b7427b03f4bc/torchvision/prototype/transforms/_auto_augment.py#L67-L68

I'll try to refactor so we can use the new `query_chw`.

Reply:
The AutoAugment transforms are an outlier of how `fill` is handled. Internally they call `F.affine`. Our `RandomAffine` does the same but only checks the values statically with

https://github.com/pytorch/vision/blob/1a1fea3453c6f329168f91c7cdfe95e043661256/torchvision/prototype/transforms/_geometry.py#L211-L213

Inside `F.affine` we do 

https://github.com/pytorch/vision/blob/1a1fea3453c6f329168f91c7cdfe95e043661256/torchvision/prototype/transforms/functional/_geometry.py#L374-L384

but only for tensor images

https://github.com/pytorch/vision/blob/1a1fea3453c6f329168f91c7cdfe95e043661256/torchvision/prototype/transforms/functional/_geometry.py#L412-L413

Reply:
@vfdev-5 has changed quite a few things to handle fill easier. Do we still need this handling or can we move it elsewhere?

Reply:
I think we can get rid of the special handling in the auto augment transforms entirely. Both there and in `F.affine` don't do anything for PIL images. The only difference on the surface is that the auto augment version expands `fill` to match the number of channels in case we get a scalar

https://github.com/pytorch/vision/blob/1a1fea3453c6f329168f91c7cdfe95e043661256/torchvision/prototype/transforms/_auto_augment.py#L70-L71

which is not done for `F.affine`

https://github.com/pytorch/vision/blob/1a1fea3453c6f329168f91c7cdfe95e043661256/torchvision/prototype/transforms/functional/_geometry.py#L383

But that is only a surface difference, since later in the kernel we do 

https://github.com/pytorch/vision/blob/1a1fea3453c6f329168f91c7cdfe95e043661256/torchvision/transforms/functional_tensor.py#L560

`fill.expand_as(image)` will broadcast to the right shape, since a singleton dimension is broadcastable with any number of channels.

Reply:
Regardless of the image type and the (valid) type of `fill`, whether or not we call `AutoAugment._parse_fill` makes no difference:

```py
import functools
import itertools

import torch
from torchvision.prototype import features, transforms
from torchvision.prototype.transforms import functional as F
from torchvision.prototype.transforms._utils import get_chw

affine = functools.partial(F.affine, angle=30, shear=[0, 0], translate=[0, 0], scale=1)

image_tensor = torch.rand(3, 16, 16)
image_pil = F.to_image_pil(image_tensor)
image_feature = features.Image(image_tensor)

for image, fill_plain in itertools.product(
    [image_tensor, image_pil, image_feature],
    [None, 0, 0.5, [0.1, 0.5, 0.9], (0.1, 0.5, 0.9)],
):
    output_plain = affine(image, fill=fill_plain)

    num_channels, _, _ = get_chw(image)
    fill_aa = transforms.AutoAugment(fill=fill_plain)._parse_fill(image, num_channels)
    output_aa = affine(image, fill=fill_aa)

    if fill_aa != fill_plain:
        print(f"Mismatch for {type(image)}: fill_plain={fill_plain}, fill_aa={fill_aa}")
    torch.testing.assert_close(F.to_image_tensor(output_aa), F.to_image_tensor(output_plain), atol=0, rtol=0)
```

```
Mismatch for <class 'torch.Tensor'>: fill_plain=0, fill_aa=[0.0, 0.0, 0.0]
Mismatch for <class 'torch.Tensor'>: fill_plain=0.5, fill_aa=[0.5, 0.5, 0.5]
Mismatch for <class 'torch.Tensor'>: fill_plain=(0.1, 0.5, 0.9), fill_aa=[0.1, 0.5, 0.9]
Mismatch for <class 'torchvision.prototype.features._image.Image'>: fill_plain=0, fill_aa=[0.0, 0.0, 0.0]
Mismatch for <class 'torchvision.prototype.features._image.Image'>: fill_plain=0.5, fill_aa=[0.5, 0.5, 0.5]
Mismatch for <class 'torchvision.prototype.features._image.Image'>: fill_plain=(0.1, 0.5, 0.9), fill_aa=[0.1, 0.5, 0.9]
```

We should remove.

Reply:
Yes, I think you can remove AA._parse_fill, `F.affine` and `Image.affine` should be able to handle channels expansion.